LoopKit/GlucoseValue.swift

```diff
@@ -13,6 +13,48 @@ import HealthKit
 public protocol GlucoseValue: SampleValue {
 }
 
+struct SimpleGlucoseValue: Equatable, GlucoseValue {
+    public let startDate: Date
+    public let endDate: Date
+    public let quantity: HKQuantity
+
+    public init(startDate: Date, endDate: Date? = nil, quantity: HKQuantity) {
+        self.startDate = startDate
+        self.endDate = endDate ?? startDate
+        self.quantity = quantity
+    }
+
+    public init(_ glucoseValue: GlucoseValue) {
+        self.startDate = glucoseValue.startDate
+        self.endDate = glucoseValue.endDate
+        self.quantity = glucoseValue.quantity
+    }
+}
+
+extension SimpleGlucoseValue: Codable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.startDate = try container.decode(Date.self, forKey: .startDate)
+        self.endDate = try container.decode(Date.self, forKey: .endDate)
+        self.quantity = HKQuantity(unit: HKUnit(from: try container.decode(String.self, forKey: .quantityUnit)),
```

Comment:
While this is probably okay in practice, I think that `HKUnit(from:) ` will throw an `invalidArgumentException` (which is an `NSException`, not a `Swift.Error`) if the string passed is invalid.  I don't know how bullet-proof you wish to make this, but we may need to add a facility for catching ObjC exceptions (e.g. https://stackoverflow.com/questions/32758811/catching-nsexception-in-swift). Thoughts?